internal/services/dataprotection/azurerm_data_protection_backup_policy_kubernetes_cluster_resource.go

```diff
@@ -0,0 +1,700 @@
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DataProtection.BackupPolicyClient
+
+			id, err := backuppolicies.ParseBackupPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := BackupPolicyKubernatesClusterModel{
+				Name:              id.BackupPolicyName,
+				ResourceGroupName: id.ResourceGroupName,
+				VaultName:         id.BackupVaultName,
+			}
+
+			if model := resp.Model; model != nil {
+				if properties, ok := model.Properties.(backuppolicies.BackupPolicy); ok {
+					state.DefaultRetentionRule = flattenBackupPolicyKubernetesClusterDefaultRetentionRule(&properties.PolicyRules)
+					state.RetentionRule = flattenBackupPolicyKubernetesClusterRetentionRules(&properties.PolicyRules)
+					state.BackupRepeatingTimeIntervals = flattenBackupPolicyKubernetesClusterBackupRuleArray(&properties.PolicyRules)
+					state.TimeZone = flattenBackupPolicyKubernetesClusterBackupTimeZone(&properties.PolicyRules)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r DataProtectionBackupPolicyKubernatesClusterResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.DataProtection.BackupPolicyClient
+
+			id, err := backuppolicies.ParseBackupPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if _, err := client.Delete(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandBackupPolicyKubernetesClusterAzureBackupRuleArray(input []string, timeZone string, taggingCriteria *[]backuppolicies.TaggingCriteria) []backuppolicies.BasePolicyRule {
+	results := make([]backuppolicies.BasePolicyRule, 0)
+	results = append(results, backuppolicies.AzureBackupRule{
+		Name: "BackupIntervals",
+		DataStore: backuppolicies.DataStoreInfoBase{
+			DataStoreType: backuppolicies.DataStoreTypesOperationalStore,
+			ObjectType:    "DataStoreInfoBase",
+		},
+		BackupParameters: &backuppolicies.AzureBackupParams{
+			BackupType: "Incremental",
+		},
+		Trigger: backuppolicies.ScheduleBasedTriggerContext{
+			Schedule: backuppolicies.BackupSchedule{
+				RepeatingTimeIntervals: input,
+				TimeZone:               pointer.To(timeZone),
+			},
+			TaggingCriteria: *taggingCriteria,
+		},
+	})
+
+	return results
+}
+
+func expandBackupPolicyKubernetesClusterDefaultRetentionRule(input []DefaultRetentionRule) backuppolicies.BasePolicyRule {
+	results := backuppolicies.AzureRetentionRule{}
+	if len(input) == 0 {
+		return results
+	}
```

Comment:
if `len(input)==0`, we should not return an empty rule. In this case, no rule should be added to the `policyRules` list outside this method?

Reply:
Thanks for your feedback, fixed.